Tasks/S3Download/DownloadTaskParameters.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { AWSConnectionParameters } from 'sdkutils/awsConnectionParameters'
+import tl = require('vsts-task-lib/task')
+
+export class TaskParameters {
+    // options for Server-side encryption Key Management
+    public static readonly noneOrAWSManagedKeyValue: string = 'noneOrAWSManaged'
+    public static readonly customerManagedKeyValue: string = 'customerManaged'
+
+    public static readonly aes256AlgorithmValue: string = 'AES256'
+
+    public awsConnectionParameters: AWSConnectionParameters
```

Comment:
A more idiomatic way to do this would be:

```typescript
export const noneOrAWSManagedKeyValue = 'noneOrAWSManaged'
...
export interface TaskParameters {
    readonly bucketName: string
    readonly sourceFolder: string
    ...
}

export function build(): TaskParameters {
    const parameters: TaskParameters = {
        awsConnectionParameters: new AWSConnectionParameters(),
        ...
    }

    if (parameters.keyManagement === customerManagedKeyValue) {
        ...
    }

    return parameters
}
```

Reply:
This is how all of them are (not sure why git didn't realize this was a rename), although anonymous classes are OK why do you say this is more idiomatic?

Reply:
changed